rails_event_store_active_record/lib/rails_event_store_active_record/mappers/yaml.rb

```diff
@@ -0,0 +1,31 @@
+module RailsEventStoreActiveRecord
+  module Mappers
+    class YAML
+      def initialize
+        @serializer = ::YAML
+      end
+
+      def build_event_payload(event)
+        {
+          id:         event.event_id,
+          data:       serializer.dump(event.data),
+          metadata:   serializer.dump(event.metadata),
+          event_type: event.class,
+        }
```

Comment:
I would keep here hash, because it what `Event.create!` is accepting, even if I will make here close struct or object, it will be casted to_h. On begining I created event record here, but moved it back to repository

Reply:
@andrzejsliwa nah, you can assign one-by-one to ActiveRecord `Event`:

```ruby
def event_to_db(event)
  db_event = Event.new
  db_event.id = mapper.get_id(event)
  ...
end

or one by one to a hash in this persistence layer, but I would prefer keeping `Hash` out of this mapper layer.

Reply:
@paneq take a look again